serviceuser.go

```diff
@@ -160,6 +161,23 @@ func (su *ServiceUser) waitUntilReady() error {
 	return nil
 }
 
+// Connect connects to the server at the given "host:port". Either Connect or
+// SetConn must be before calling CStore, etc.
+func (su *ServiceUser) ConnectWithTimeout(serverAddr string, timeout time.Duration) error {
+	if su.status != serviceUserInitial {
+		panic(fmt.Sprintf("dicom.serviceUser: Connect called with wrong state: %v", su.status))
+	}
+	conn, err := net.DialTimeout("tcp", serverAddr, timeout)
+	if err != nil {
+		dicomlog.Vprintf(0, "dicom.serviceUser: Connect(%s): %v", serverAddr, err)
+		su.disp.downcallCh <- stateEvent{event: evt17, pdu: nil, err: err}
+		return fmt.Errorf("net.DialTimeout: %w", err)
+	} else {
+		su.disp.downcallCh <- stateEvent{event: evt02, pdu: nil, err: nil, conn: conn}
+	}
+	return nil
+}
+
 // Connect connects to the server at the given "host:port". Either Connect or
 // SetConn must be before calling CStore, etc.
 func (su *ServiceUser) Connect(serverAddr string) {
```

Comment:
The Connect and ConnectWithTimeout are mostly duplicated. We could turn this old function into

```
func (su *ServiceUser) Connect(serverAddr string) {
    return su.ConnectContext(context.Background(), serverAddr)
}
```

Reply:
Fixed at f6f6d07704b9323da4735839ed0f1eee60b0f1f6